include/seqan3/core/bit_manipulation.hpp

```diff
@@ -77,30 +79,159 @@ constexpr size_t next_power_of_two(size_t n)
     return n + 1;
 }
 
-/*!\brief Returns the position of the most significant bit (counting from right to left).
+/*!\brief Returns the number of 1-bits.
+* \ingroup core
+*
+* \param[in] n An unsigned integer.
+*
+* \returns The number of 1-bits.
+*
+* ### Example
+*
+* \include test/snippet/core/detail/popcount.cpp
+*
+* ### Exception
+*
+* No-throw guarantee.
+*
+* ### Thread-safety
+*
+* Thread safe.
+*
+* ### Complexity
+*
+* Constant.
```

Comment:
sdsl bits count does some magic bits trickery to get constant time? Or is there somewhere a fall back to O(n) bits?

Reply:
I looked a bit more into it and from what I've seen and read, it should be constant with all the lookup tables.
To be really sure, I would have to look it up in Donald Knuth's *The Art of Computer Programming*.

I can definitely say that it's never scanning the bits from left to right or such.